pages/_app.js

```diff
@@ -4,8 +4,44 @@ import styled from "styled-components";
 import Navbar from "@/components/Navbar";
 import Header from "@/components/Header";
 import { useState } from "react";
+import FormModal from "@/components/FormModal";
 export default function App({ Component, pageProps }) {
   const [localGameData, setLocalGameData] = useState([]);
+  const [modal, setModal] = useState({
+    isVisible: false,
+    isEdit: false,
+    isSubmit: false,
+    game: null,
+  });
+
+  function toggleShowModal(string, game) {
+    string === "edit" &&
+      setModal({
+        isVisible: true,
+        isEdit: true,
+        game: game,
+        isSubmit: false,
+      });
+
+    string === "create" &&
+      setModal({ isVisible: true, isEdit: false, isSubmit: false, game: null });
+
+    string === "submit" &&
+      setModal({
+        isVisible: true,
+        isEdit: modal.isEdit,
+        isSubmit: true,
+        game: null,
+      });
+
+    string === "close" &&
+      setModal({
+        isVisible: false,
+        isEdit: false,
+        isSubmit: false,
+        game: null,
+      });
```

Comment:
a very nice solution here. If you want to improve the readability and performance a bit, you could use early return statements. That would stop the function execution when the first match for the "string" is reached. Here, all cases are still checked, even after one is already a match. A